keps/sig-network/2091-cluster-network-policy/README.md

```diff
@@ -0,0 +1,1161 @@
+```
+
+Note: the above ClusterNetworkPolicy can also be written in the following fashion:
+```yaml
+apiVersion: netpol.networking.k8s.io/v1alpha1
+kind: ClusterNetworkPolicy
+spec:
+  priority: 50
+  - appliedTo:
+     namespaceSelector:
+       matchExpressions: {key: "tenant"; operator: Exists}
+    ingress:
+     - action: Pass
+       from:
+       - namespaces:
+          scope: samelabels
+          labels:
+            - tenant
+     - action: Deny   # Deny everything else other than same tenant traffic
+       from:
+       - namespaceSelector: {}
+```
+The difference is that in the first case, traffic within tenant Namespaces will fall
+through, and be evaluated against lower-priority ClusterNetworkPolicies, and then
+NetworkPolicies. In the second case, the matching packet will skip all ClusterNetworkPolicy
+evaluation (except for ClusterNetworkPolicy priority=baseline), and only match
+against NetworkPolicy rules in the cluster. In other words, the second ClusterNetworkPolicy
+specifies intra-tenant traffic must be delegated to the tenant Namespace owners.
+
+#### Sample spec for Story 5: Cluster Wide Default Guardrails
+
+```yaml
+apiVersion: netpol.networking.k8s.io/v1alpha1
+kind: ClusterNetworkPolicy
+spec:
+  priority: baseline
+  - appliedTo:
+     namespaceSelector:
+       matchExpressions: {key: "tenant"; operator: Exists}
+    ingress:
+     - action: Deny   # zero-trust cluster default security posture
+       from:
+       - namespaceSelector: {}
+```
+
+### Test Plan
+
+- Add e2e tests for ClusterNetworkPolicy resource
+  - Ensure `Pass` rules are delegated and are not subject to CNP rules.
+  - Ensure `Deny` rules drop traffic.
+  - Ensure `Allow` rules allow traffic.
+  - Ensure that in stacked ClusterNetworkPolicies/K8s NetworkPolicies, precedence is maintained
+    as per the `priority` set in CNP.
+- e2e test cases must cover ingress and egress rules
+- e2e test cases must cover port-ranges, named ports, integer ports etc
+- e2e test cases must cover various combinations of `workloadSelector` in `appliedTo` and ingress/egress rules
+- e2e test cases must cover various combinations of `namespaceSelector` in `appliedTo`
+- e2e test cases must cover various combinations of `namespaces` in ingress/egress rules
+  - Ensure that namespace matching strategies work as expected
+- Add unit tests to test the validation logic which shall be introduced for cluster-scoped policy resources
+  - Ensure that `self` field cannot be set along with `selector` within `namespaces`
+  - Test cases for fields which are shared with NetworkPolicy, like `endPort` etc.
+  - Ensure no two policies can be set with same `priority`
+- Ensure that only administrators or assigned roles can create/update/delete cluster-scoped policy resources
+
+### Graduation Criteria
+
+#### Alpha to Beta Graduation
+
+- Gather feedback from developers and surveys
+- At least 2 CNI provider must provide the implementation for the complete set
+  of alpha features
```

Comment:
We should discuss the numbers here. We used "2" for NetworkPolicy port ranges. I don't think we should be using the same criteria for something this large.